runtime/lua/vim/treesitter/query.lua

```diff
@@ -0,0 +1,176 @@
+local a = vim.api
+local lang = require'vim.treesitter.language'
+
+-- query: pattern matching on trees
+-- predicate matching is implemented in lua
+local Query = {}
+Query.__index = Query
+
+local M = {}
+
+local magic_prefixes = {['\\v']=true, ['\\m']=true, ['\\M']=true, ['\\V']=true}
+local function check_magic(str)
+  if string.len(str) < 2 or magic_prefixes[string.sub(str,1,2)] then
+    return str
+  end
+  return '\\v'..str
+end
+
+-- Some treesitter grammars extend others.
+-- We can use that to import the queries of the base language
+M.base_language_map = {
```

Comment:
Should have a function to add / modify these probably. Just in case we change the structure later.

Reply:
And with that, then maybe hide this map as well (like, just make it a local variable and access it with some helper functions outside of the module).